lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -1821,18 +1821,30 @@ static void fixAvailabilityForDecl(SourceRange ReferenceRange, const Decl *D,
   if (TypeChecker::diagnosticIfDeclCannotBePotentiallyUnavailable(D).has_value())
     return;
 
-  if (getActiveAvailableAttribute(D, Context)) {
-    // For QoI, in future should emit a fixit to update the existing attribute.
-    return;
-  }
-
   // For some declarations (variables, enum elements), the location in concrete
   // syntax to suggest the Fix-It may differ from the declaration to which
   // we attach availability attributes in the abstract syntax tree during
   // parsing.
   const Decl *ConcDecl = concreteSyntaxDeclForAvailableAttribute(D);
 
   DescriptiveDeclKind KindForDiagnostic = ConcDecl->getDescriptiveKind();
+
+  if (auto Attr = getActiveAvailableAttribute(D, Context)) {
+    if (Attr->isUnconditionallyUnavailable()) {
+      return;
+    }
+    SourceManager &SM = Context.SourceMgr;
+    auto Version = SM.extractText(Lexer::getCharSourceRangeFromSourceRange(
+                                      SM, Attr->IntroducedRange))
+                       .str();
```

Comment:
Thank you for your review.
`Attr->Introduced->getAsString()`works well in most cases, but due to canonicalizePlatformVersion transforming the version number, when encountering`@available(macOS, introduced: 10.16)`, `Attr->Introduced->getAsString()`will return **11** instead of **10.16**.  This may cause more confusion than leading zeros.

```cpp
llvm::VersionTuple swift::canonicalizePlatformVersion(
    PlatformKind platform, const llvm::VersionTuple &version) {

  // Canonicalize macOS version for macOS Big Sur to treat
  // 10.16 as 11.0.
  if (platform == PlatformKind::macOS ||
      platform == PlatformKind::macOSApplicationExtension) {
    return llvm::Triple::getCanonicalVersionForOS(llvm::Triple::MacOSX,
                                                  version);
  }

  return version;
}
```

Reply:
I think this is okay because technically there *was* no macOS 10.16. The compiler is rewriting something less correct into something more correct.